### PR TITLE
[GitHubHacktoberfest] 2020 (and beyond)

### LINKED ISSUE #5545: Hacktoberfest 2020 Badge
:clipboard: **Description**

The current Hacktoberfest badge is for Hacktoberfest 2019. There's no option to change the year - `2019` is hardcoded.

Screenshot 2020-09-15 at 10 55 34 AM

GitHub Hacktoberfest combined status

:link: **Data**

This badge was already implemented in 2019, I hope the APIs exist with you.

:microphone: **Motivation**

Having a Hacktoberfest countdown badge will be great as [Hacktoberfest 2020](https://hacktoberfest.digitalocean.com/) is approaching us.

### PR DESCRIPTION
**Pull request by calebcartwright**
Resolves #5545 

I decided to remove the 2019 hardcoding, but also constrain the route param for the year to a set of allowed values that only includes 2019 and 2020 (so should this be merged, next year we'd need to add 2021). I also had to take into account what the badge message should be for an upcoming/future Hacktoberfest, and elected to show a countdown of days until Hacktoberfest would begin. It didn't make sense to me pre-Hacktoberfest kickoff for the badge to show the number of days remaining until the end of October (for example, that'd currently have looked something like ), so instead we have:

https://shields-staging-pr-5549.herokuapp.com/github/hacktoberfest/2020/badges/shields

https://shields-staging-pr-5549.herokuapp.com/github/hacktoberfest/2019/badges/shields

One thing I feel is nice with this is that the badge will serve a dual purpose: folks could go ahead and add the 2020 badge to their projects today and it will act as a countdown, and then on October first it will automatically convert to the issue/PR and days remaining count for them.

In this pre-Hacktoberfest start for specified year scenario, I also decided to skip running the query since there's not really any point trying to determine how many PRs were opened during the month of October before October has started. One side effect of this is that if a user adds the 2020 hacktoberfest badge pre-October 1st but has a typo in their org/repo name, then the countdown days will still show without telling the user the repo wasn't found.. until October 1st at which point it'll change to

### CODE REVIEW

File: services/github/github-hacktoberfest.service.js [hunk 1]
```diff
@@ -65,8 +65,8 @@ module.exports = class GithubHacktoberfestCombinedStatus extends GithubAuthV4Ser
 
   static get route() {
     return {
-      base: 'github/hacktoberfest/2019',
-      pattern: ':user/:repo',
+      base: 'github/hacktoberfest',
+      pattern: ':year(2019|2020)/:user/:repo',
```
Comment (calebcartwright) [inline services/github/github-hacktoberfest.service.js:69]: As discussed when we first added this badge last year, [we want to keep the year as an explicit part of the route](https://github.com/badges/shields/pull/4109#issuecomment-537642828). I wasn't sure if we wanted to completely open this up just yet, or keep it constrained to a couple years, but obviously opted for the latter.

File: services/github/github-hacktoberfest.service.js [hunk 2]
```diff
@@ -113,12 +116,24 @@ module.exports = class GithubHacktoberfestCombinedStatus extends GithubAuthV4Ser
...
+    suggestedIssueCount,
+    contributionCount,
+    daysLeft,
+    daysToStart,
+    year,
+    hasStarted = true,
+  }) {
+    if (!hasStarted) {
+      return {
+        message: `${daysToStart} ${maybePluralize('day', daysToStart)} til kickoff!`,
```
Comment (calebcartwright) [inline services/github/github-hacktoberfest.service.js:129]: I wasn't really sure what do with the message here, definitely open to suggestions/alternatives

File: services/github/github-hacktoberfest.service.js [hunk 3]
```diff
@@ -59,66 +59,71 @@ const queryParamSchema = Joi.object({
...
+    daysToStart,
+    year,
+    hasStarted = true,
+  }) {
+    if (!hasStarted) {
+      return {
+        message: `${daysToStart} ${maybePluralize(
+          'day',
+          daysToStart
+        )} til kickoff!`,
```
Comment (PyvesB) [inline services/github/github-hacktoberfest.service.js:119]: This spelling caught my eye. Reading on various Internet resources, it seems that `till` should be preferred.

https://www.grammarly.com/blog/until-till-til/
https://writingexplained.org/til-or-till-difference
https://www.merriam-webster.com/words-at-play/should-you-use-until-or-till-or-til

**Review by PyvesB — COMMENTED** [review]
Thanks for picking this up, mostly looks good, simply a spelling comment inline. 👍🏻

File: services/github/github-hacktoberfest.service.js (hunk 3 above)
Comment (calebcartwright) [reply services/github/github-hacktoberfest.service.js:119]: TIL I learned that til... :laughing:

**Comment by PyvesB** [comment]
> In this pre-Hacktoberfest start for specified year scenario, I also decided to skip running the query since there's not really any point trying to determine how many PRs were opened during the month of October before October has started. One side effect of this is that if a user adds the 2020 hacktoberfest badge pre-October 1st but has a typo in their org/repo name, then the countdown days will still show without telling the user the repo wasn't found.. until October 1st at which point it'll change to 

Didn't mention it in the first pass, but I don't think that side effect is a big problem.

**Review by PyvesB — APPROVED** [review]

**Pull request merged**